include/seqan3/core/type_list/traits.hpp

```diff
@@ -447,6 +472,32 @@ template <ptrdiff_t n, seqan3::detail::TypeList list_t>
 //!\endcond
 using split_after = decltype(detail::split_after<n>(type_list<>{}, list_t{}));
 
+
+/*!\brief Checks if unary predicate `func` returns `true` for all types in the type list.
+ * \tparam list_t The type list for which to check the given predicate on every type.
+ * \tparam unary_function_t The type of the unary predicate.
+ * \param[in] func The unary predicate to call for every type in the type list.
+ *
+ * \returns `true` if the predicate returns `true` for all types, otherwise `false`.
+ *
+ * \details
+ *
+ * This function behaves like std::all_of but on types. The type will be wrapped into std::type_identity and passed as
+ * argument. The invocation(s) will be done without any loop.
+ *
+ * This function can handle types which are incomplete, forward declared and not std::Semiregular.
+ * The result will be `false` if invocation of the unary predicate on any of the given types fails.
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `size<list_t>` (linear).
+ */
+template <seqan3::detail::TypeList list_t, typename unary_function_t>
+constexpr bool all_of(unary_function_t func)
+{
+    return detail::all_of(list_t{}, func);
+}
+
```

Comment:
>Currently all our publicly facing list/pack-traits are types or values. But this is a function which complicates the whole design 

Who said, that it only has to be types or values? There is nothing wrong with a function. I also still think this goes into a different header called algorithms. But I really don't care at the moment because these are structural issues not code issues.

>Also in the current design the trait specific arguments come first, and the type_list/pack last. This is necessary for packs and enables us to implement the type_list version in terms of the type pack version for most traits (instead of the other way around which causes more template instantiation).

Well the order is really the smallest concern. I did it like this in the first place because it is done so in the algorithms as well. And it would be even possible to implement `all_of(v1, v2, v3, fn)` using `take` and the fact that `(args, ...)` would return the last argument. Yet, that might be some additional instantiation but it is all not nested not recursive so I think performance will be a small issue on this.

Reply:
is this resolved?

Reply:
i think after the discussion in the strategy meeting all of it is resolved.